torchgen/model.py

```diff
@@ -838,9 +874,76 @@ def from_dict(
         return NativeFunctionsGroup(
             functional=functional,
             inplace=inplace,
+            mutable=mutable,
             out=out,
         )
 
+    # This function, given a "base" function of one SchemaKind, as well as a target SchemaKind,
+    # generates a new NativeFunction with the same properties, but using the target SchemaKind.
+    # We only actually generate functions for either functional or out= SchemaKinds.
+    @staticmethod
+    def generate_function(f: NativeFunction, k: SchemaKind) -> Tuple[NativeFunction, Dict[DispatchKey, Dict['OperatorName', 'BackendMetadata']]]:
+        from torchgen.api import cpp
+
+        if k == SchemaKind.functional:
+            assert f.func.kind() != SchemaKind.functional
+            gets_composite_kernel = True
+            # The new "functional" NativeFunction has:
+            # - any mutable arguments have been converted into (immutable) returns.
+            #   (if a mutable argument was not also a return, it gets converted to one)
+            # - a "functional" overload name.
+            # The default grouping logic in signature() actually already does this,
+            # so we can piggy-back off it (but we still want return names)
+            func = f.func.signature(keep_return_names=True).with_name(
+                f.func.name.remove_inplace().with_overload('functional' if not f.func.name.overload_name else f"{f.func.name.overload_name}_functional")
+            )
+        elif k == SchemaKind.out:
+            # We generate out= ops mostly just so that we can pair up NativeFunctions into groups easily,
+            # but at least today, there is no good reason to actually use them.
+            # we'll generate a dispatcher entry for them, but won't actually register any kernels for them.
+            gets_composite_kernel = False
```

Comment:
See the comment here. At least today, I don't think there's a use case for having actually implementations for these generated out= ops in core (I'd imagine that mobile will want to write their own kernels). So here we'll effectively add these ops to aten, but without any implementations.